nautobot/extras/models/relationships.py

```diff
@@ -533,6 +553,115 @@ def clean(self):
+    @classmethod
+    def required_related_objects_errors(cls, instance_or_class, output_for="ui", initial_data=None):
+        """
+        Args:
+            instance_or_class: either a model instance or class name to check its required relationships
+            output_for: either "ui" or "api" depending on usage
+            initial_data: submitted form/serializer data to validate against
+        Returns:
+            List of field error dicts if any are found
+        """
+
+        required_relationships = Relationship.objects.get_required_for_model(instance_or_class)
+        relationships_field_errors = []
+        for relation in required_relationships:
+
+            opposite_side = RelationshipSideChoices.OPPOSITE[relation.required_side]
+
+            if relation.skip_required(instance_or_class, opposite_side):
+                continue
+
+            required_model_class = getattr(relation, f"{opposite_side}_type").model_class()
+            required_model_meta = required_model_class._meta
+            cr_field_name = f"cr_{relation.slug}__{opposite_side}"
+            name_plural = instance_or_class._meta.verbose_name_plural
+            field_key = relation.slug if output_for == "api" else cr_field_name
+            field_errors = {field_key: []}
+
+            if relation.has_many(opposite_side):
+                num_required_verbose = "at least one"
+            else:
+                num_required_verbose = "a"
+
+            if not required_model_class.objects.exists():
+
+                hint = (
+                    f"You need to create {num_required_verbose} {required_model_meta.verbose_name} "
+                    f"before instantiating a {instance_or_class._meta.verbose_name}."
+                )
+
+                if output_for == "ui":
+                    try:
+                        add_url = reverse(get_route_for_model(required_model_class, "add"))
+                        hint = (
+                            f"<a target='_blank' href='{add_url}'>Click here</a> to create "
+                            f"a {required_model_meta.verbose_name}."
+                        )
+                    except NoReverseMatch:
+                        pass
+
+                elif output_for == "api":
+                    try:
+                        api_post_url = reverse(get_route_for_model(required_model_class, "list", api=True))
+                        hint = f"Create a {required_model_meta.verbose_name} by posting to {api_post_url}"
+                    except NoReverseMatch:
+                        pass
+
+                error_message = mark_safe(
+                    f"{name_plural[0].upper()}{name_plural[1:]} require "
+                    f"{num_required_verbose} {required_model_meta.verbose_name}, but no "
+                    f"{required_model_meta.verbose_name_plural} exist yet. {hint}"
+                )
+                field_errors[field_key].append(error_message)
+
+            if isinstance(initial_data, dict):
```

Comment:
When is it not a dict? What should we do in that case?

Reply:
It's `None` by default in kwargs so that this method can be used to check required relationships errors without the need to supply data.